README.md
# MYERP

## Organisation du répertoire

*   `doc` : documentation
*   `docker` : répertoire relatifs aux conteneurs _docker_ utiles pour le projet
    *   `dev` : environnement de développement
*   `racine` : code source de l'application

## Environnement de développement

Les composants nécessaires lors du développement sont disponibles via des conteneurs _docker_.
L'environnement de développement est assemblé grâce à _docker-compose_
(cf docker/dev/docker-compose.yml).

Il comporte :

*   une base de données _PostgreSQL_ contenant un jeu de données de démo (`postgresql://127.0.0.1:9032/db_myerp`)

 
## Déploiement Docker

#### Lancement

    cd docker/dev
    docker-compose up

#### Arrêt

    cd docker/dev
    docker-compose stop

#### Remise à zero

    cd docker/dev
    docker-compose stop
    docker-compose rm -v
    docker-compose up
    
#### Réinitialisation de la base de données
Le script `restart.sh` présent dans le dossier `docker/dev` permet de supprimer le container 
de la base de données et de le recréer avec le jeu de données de base.

    Important : Cette manipulation est à effectuer à chaque lancement de build Jenkins.

## Intégration continue avec Jenkins
L'intégration continue du projet se fait via Jenkins. Un fichier `Jenkinsfile` permet de configurer 
*pipeline*. Ce pipeline va :
- Récupérer les sources depuis le repository
- Lancer une commande `mvn clean install` avec les profils de tests **consumer-test** et **business-test**

### Installation du pipeline Jenkins
L'instance de Jenkins installée doit être configurée avec JDK 8, Sonarqube et Maven 3.

Une fois le pipeline créé, il suffit d'y configurer le lien vers le repository et 
de spécifier que la configuration du pipeline se fera avec unfichier *Jenkinsfile*.

### Contenu du script Jenkinsfile 
Un fichier `Jenkinsfile` permet d'automatiser les tests et leur agrégation :

Au lancement de build : 
1. Jenkins récupère le repository local
2. Une commande est lancée pour effectuer les tests avec les profils *consumer-test* et *business-test*

Le fichier `restart.sh` permet de relancer la base de données pour effectuer un build (voir **Déploiement du projet**)

### Agrégation des données de tests
Les données liées aux tests sont automatiquement intégrées, via la commande maven lancée depuis Jenkins, 
à **Sonarqube**.

## Correctifs et évolutions

*Se référer au fichier ``CHANGELOG.md``* 

CHANGELOG.md
# CHANGELOG
Ce document regroupe l'ensemble des modifications intervenus dans le projet ``MyERP``
## [Unreleased]

## [1.1.0] - 2019-06-01

### Changement
- Implémentation de la méthode `addReference()` dans `ComptabiliteManagerImpl.class`
- Implémentation de la règle de gestion RG-Compta5 dans la méthode `CheckEcritureComptableUnit()`
- Ajout d'une base de données de test via le fichier `consumerContext.xml` dans le module **Consumer**

### Correctifs
- EcritureComptable, isEquilibree() : Erreur lors de la comparaison des données causée par les décimales. Ajout de la méthode stripTrailingZeros() pour suprimer les décimales inutiles. 
- SqlContext.xml, SQLinsertListLigneEcritureComptable : Ajout d'une virgule dans la requête SQL
- Création du fichier de configuration `bootstrapContext.xml` dans le module **Business**
- EcritureComptable, getTotalCredit() : Erreur provoquée par l'appel de la méthode getDebit() au lieu de getCredit()
- EcritureComptable : Modification de l'expression régulière


myerp-business/src/test/java/com/dummy/myerp/business/impl/manager/BusinessManagerHelper.java
package com.dummy.myerp.business.impl.manager;

import com.dummy.myerp.business.impl.AbstractBusinessManager;

public class BusinessManagerHelper extends AbstractBusinessManager {

    public BusinessManagerHelper() {
        super();
    }

    public static BusinessManagerHelper getInstance() {
        return new BusinessManagerHelper();
    }
}


myerp-business/src/test/java/com/dummy/myerp/business/impl/manager/ComptabiliteManagerImplTest.java
package com.dummy.myerp.business.impl.manager;

import com.dummy.myerp.business.contrat.BusinessProxy;
import com.dummy.myerp.business.impl.TransactionManager;
import com.dummy.myerp.consumer.dao.contrat.DaoProxy;
import com.dummy.myerp.consumer.dao.impl.DaoProxyImpl;
import com.dummy.myerp.consumer.dao.impl.db.dao.ComptabiliteDaoImpl;
import com.dummy.myerp.model.bean.comptabilite.CompteComptable;
import com.dummy.myerp.model.bean.comptabilite.EcritureComptable;
import com.dummy.myerp.model.bean.comptabilite.JournalComptable;
import com.dummy.myerp.model.bean.comptabilite.LigneEcritureComptable;
import com.dummy.myerp.technical.exception.FunctionalException;
import com.dummy.myerp.technical.exception.NotFoundException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;
import org.mockito.junit.jupiter.MockitoSettings;
import org.mockito.quality.Strictness;

import javax.validation.Validator;
import java.math.BigDecimal;
import java.util.Date;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

@ExtendWith(MockitoExtension.class)
@MockitoSettings(strictness = Strictness.LENIENT)
public class ComptabiliteManagerImplTest {

    class ComptabiliterManagerTestClass extends ComptabiliteManagerImpl {
        @Override
        protected BusinessProxy getBusinessProxy() {
            return super.getBusinessProxy();
        }

        @Override
        protected DaoProxy getDaoProxy() {
            return super.getDaoProxy();
        }

        @Override
        protected TransactionManager getTransactionManager() {
            return super.getTransactionManager();
        }

        @Override
        protected Validator getConstraintValidator() {
            return super.getConstraintValidator();
        }
    }

    private EcritureComptable vEcritureComptable;

    private ComptabiliteManagerImpl vComptabiliteManager;

    private ComptabiliterManagerTestClass vComptabiliteManagerMock;
    private ComptabiliterManagerTestClass vComptabiliteManagerSpy;

    private ComptabiliteDaoImpl vComptabiliteDaoMock;
    private ComptabiliteDaoImpl vComptabiliteDaoSpy;

    private DaoProxyImpl vDaoProxyMock;
    private DaoProxyImpl vDaoProxySpy;

    @BeforeEach
    public void setUpBeforeEach() {
        vComptabiliteManagerMock = mock(ComptabiliterManagerTestClass.class);
        vComptabiliteManagerSpy = spy(vComptabiliteManagerMock);

        vComptabiliteDaoMock = mock(ComptabiliteDaoImpl.class);
        vComptabiliteDaoSpy = spy(vComptabiliteDaoMock);

        vDaoProxyMock = mock(DaoProxyImpl.class);
        vDaoProxySpy = spy(vDaoProxyMock);

        vComptabiliteManager = new ComptabiliteManagerImpl();

        vEcritureComptable = new EcritureComptable();
        vEcritureComptable.setId(1);
        vEcritureComptable.setJournal(new JournalComptable("AC", "Achat"));
        vEcritureComptable.setDate(new Date());
        vEcritureComptable.setLibelle("Libelle");
        vEcritureComptable.setReference("AC-2019/00001");
    }

    @Test
    public void checkEcritureComptable() throws NotFoundException {
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(2),
                null, null,
                new BigDecimal(123)));

        doReturn(vDaoProxyMock).when(vComptabiliteManagerSpy).getDaoProxy();
        doReturn(vComptabiliteDaoSpy).when(vDaoProxySpy).getComptabiliteDao();
        doReturn(null).when(vComptabiliteDaoSpy).getEcritureComptableByRef(anyString());

        assertDoesNotThrow(() -> vComptabiliteManagerSpy.checkEcritureComptable(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableWithExistingRefWithoutId() throws FunctionalException, NotFoundException {
        EcritureComptable vEcritureComptableSample = new EcritureComptable();
        vEcritureComptableSample.setJournal(new JournalComptable("AC", "Achat"));
        vEcritureComptableSample.setDate(new Date());
        vEcritureComptableSample.setLibelle("Libelle");
        vEcritureComptableSample.setReference("AC-2019/00001");

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptable(vEcritureComptableSample));
    }

    @Test
    public void checkEcritureComptableContextWithoutSameId() throws FunctionalException, NotFoundException {
        EcritureComptable vEcritureComptableSample = new EcritureComptable();
        vEcritureComptableSample.setId(2);
        vEcritureComptableSample.setJournal(new JournalComptable("AC", "Achat"));
        vEcritureComptableSample.setDate(new Date());
        vEcritureComptableSample.setLibelle("Libelle");
        vEcritureComptableSample.setReference("AC-2019/00001");

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptable(vEcritureComptableSample));
    }

    @Test
    public void checkEcritureComptableUnitViolation() throws Exception {
        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableUnitRG2() throws Exception {
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(2),
                null, null,
                new BigDecimal(1234)));

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableUnitRG3() throws Exception {
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableUnitRG3_NombreLignes() throws Exception {
        vEcritureComptable.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableUnitRG5WithWrongYear() throws Exception {
        vEcritureComptable.setReference("AAC-2018/00001");

        vComptabiliteManager.addReference(vEcritureComptable);

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkEcritureComptableUnitRG5WithWrongCode() throws Exception {
        vEcritureComptable.setReference("AAC-2019/00001");

        vComptabiliteManager.addReference(vEcritureComptable);

        assertThrows(FunctionalException.class, () -> vComptabiliteManager.checkEcritureComptableUnit(vEcritureComptable));
    }

    @Test
    public void checkAddReferenceAsFirst() throws Exception {
        vEcritureComptable.setReference(null);

        vComptabiliteManager.addReference(vEcritureComptable);

        assertEquals("AC-2019/00001", vEcritureComptable.getReference());
    }

    @Test
    public void checkAddReferenceAsNew() throws Exception {
        EcritureComptable vNewEcritureComptable = new EcritureComptable();
        vNewEcritureComptable.setJournal(new JournalComptable("AC", "Achat"));
        vNewEcritureComptable.setDate(new Date());
        vNewEcritureComptable.setLibelle("Libelle");

        vComptabiliteManager.addReference(vNewEcritureComptable);

        assertEquals("AC-2019/00001", vNewEcritureComptable.getReference());
    }
}


myerp-model/src/main/java/com/dummy/myerp/model/bean/comptabilite/EcritureComptable.java
package com.dummy.myerp.model.bean.comptabilite;


import org.apache.commons.lang.StringUtils;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;



/**
 * Bean représentant une Écriture Comptable
 */
public class EcritureComptable {

    // ==================== Attributs ====================
    /** The Id. */
    private Integer id;
    /** Journal comptable */
    @NotNull private JournalComptable journal;
    /** The Reference. */
    @Pattern(regexp = "\\w{1,5}-\\d{4}/\\d{5}")
    private String reference;
    /** The Date. */
    @NotNull private Date date;

    /** The Libelle. */
    @NotNull
    @Size(min = 1, max = 200)
    private String libelle;

    /** La liste des lignes d'écriture comptable. */
    @Valid
    @Size(min = 2)
    private final List<LigneEcritureComptable> listLigneEcriture = new ArrayList<>();


    // ==================== Getters/Setters ====================
    public Integer getId() {
        return id;
    }
    public void setId(Integer pId) {
        id = pId;
    }
    public JournalComptable getJournal() {
        return journal;
    }
    public void setJournal(JournalComptable pJournal) {
        journal = pJournal;
    }
    public String getReference() {
        return reference;
    }
    public void setReference(String pReference) {
        reference = pReference;
    }
    public Date getDate() {
        return date;
    }
    public void setDate(Date pDate) {
        date = pDate;
    }
    public String getLibelle() {
        return libelle;
    }
    public void setLibelle(String pLibelle) {
        libelle = pLibelle;
    }
    public List<LigneEcritureComptable> getListLigneEcriture() {
        return listLigneEcriture;
    }

    /**
     * Calcul et renvoie le total des montants au débit des lignes d'écriture
     *
     * @return {@link BigDecimal}, {@link BigDecimal#ZERO} si aucun montant au débit
     */
    public BigDecimal getTotalDebit() {
        BigDecimal vRetour = BigDecimal.ZERO;
        for (LigneEcritureComptable vLigneEcritureComptable : listLigneEcriture) {
            if (vLigneEcritureComptable.getDebit() != null) {
                vRetour = vRetour.add(vLigneEcritureComptable.getDebit());
            }
        }
        return vRetour.stripTrailingZeros();
    }

    /**
     * Calcul et renvoie le total des montants au crédit des lignes d'écriture
     *
     * @return {@link BigDecimal}, {@link BigDecimal#ZERO} si aucun montant au crédit
     */
    public BigDecimal getTotalCredit() {
        BigDecimal vRetour = BigDecimal.ZERO;
        for (LigneEcritureComptable vLigneEcritureComptable : listLigneEcriture) {
            if (vLigneEcritureComptable.getCredit() != null) {
                vRetour = vRetour.add(vLigneEcritureComptable.getCredit());
            }
        }
        return vRetour.stripTrailingZeros();
    }

    /**
     * Renvoie si l'écriture est équilibrée (TotalDebit = TotalCrédit)
     * @return boolean
     */
    public boolean isEquilibree() {
        boolean vRetour = this.getTotalDebit().equals(getTotalCredit());
        return vRetour;
    }

    // ==================== Méthodes ====================
    @Override
    public String toString() {
        final StringBuilder vStB = new StringBuilder(this.getClass().getSimpleName());
        final String vSEP = ", ";
        vStB.append("{")
                .append("id=").append(id)
                .append(vSEP).append("journal=").append(journal)
                .append(vSEP).append("reference='").append(reference).append('\'')
                .append(vSEP).append("date=").append(date)
                .append(vSEP).append("libelle='").append(libelle).append('\'')
                .append(vSEP).append("totalDebit=").append(this.getTotalDebit().toPlainString())
                .append(vSEP).append("totalCredit=").append(this.getTotalCredit().toPlainString())
                .append(vSEP).append("listLigneEcriture=[\n")
                .append(StringUtils.join(listLigneEcriture, "\n")).append("\n]")
                .append("}");
        return vStB.toString();
    }
}


myerp-business/src/test-business/java/com/dummy/myerp/testbusiness/business/ComptabiliteManagerIntegrationTest.java
package com.dummy.myerp.testbusiness.business;

import com.dummy.myerp.business.contrat.manager.ComptabiliteManager;
import com.dummy.myerp.model.bean.comptabilite.CompteComptable;
import com.dummy.myerp.model.bean.comptabilite.EcritureComptable;
import com.dummy.myerp.model.bean.comptabilite.JournalComptable;
import com.dummy.myerp.model.bean.comptabilite.LigneEcritureComptable;
import com.dummy.myerp.technical.exception.FunctionalException;
import org.junit.jupiter.api.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Date;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class ComptabiliteManagerIntegrationTest {

    private ComptabiliteManager vComptabiliteManager = BusinessTestCase.getBusinessProxy().getComptabiliteManager();
    private EcritureComptable vEcritureComptableSample;

    public ComptabiliteManagerIntegrationTest() {
        super();
    }

    @BeforeEach
    public void setupBeforeEach() {
        vEcritureComptableSample = new EcritureComptable();
        vEcritureComptableSample.setJournal(new JournalComptable("AC", "Achat"));
        vEcritureComptableSample.setDate(Date.from(Instant.now()));
        vEcritureComptableSample.setLibelle("Libelle");
        vEcritureComptableSample.setReference("AC-2019/00001");

        vEcritureComptableSample.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(1),
                null, new BigDecimal(123),
                null));
        vEcritureComptableSample.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(2),
                null, null,
                new BigDecimal(123)));
    }

    @Test
    @Order(1)
    public void insertEcritureComptableTest() throws FunctionalException {
        vComptabiliteManager.insertEcritureComptable(vEcritureComptableSample);
    }

    @Test
    @Order(2)
    public void updateEcritureComptable() throws FunctionalException {
        vEcritureComptableSample.setLibelle("Un nouveau libelle de test");
        vComptabiliteManager.updateEcritureComptable(vEcritureComptableSample);
    }

    @Test
    @Order(3)
    public void deleteEcritureComptableTest() {
        vComptabiliteManager.deleteEcritureComptable(vComptabiliteManager.getListEcritureComptable().size() - 1);
    }
}


myerp-consumer/src/test-consumer/java/com/dummy/myerp/consumer/dao/impl/db/dao/ComptabiliteDaoImplTest.java
package com.dummy.myerp.consumer.dao.impl.db.dao;

import com.dummy.myerp.consumer.dao.impl.DaoProxyImpl;
import com.dummy.myerp.model.bean.comptabilite.CompteComptable;
import com.dummy.myerp.model.bean.comptabilite.EcritureComptable;
import com.dummy.myerp.model.bean.comptabilite.JournalComptable;
import com.dummy.myerp.model.bean.comptabilite.LigneEcritureComptable;
import com.dummy.myerp.technical.exception.NotFoundException;
import org.junit.jupiter.api.*;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.junit.jupiter.MockitoExtension;
import org.mockito.junit.jupiter.MockitoSettings;
import org.mockito.quality.Strictness;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Date;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

@ExtendWith(MockitoExtension.class)
@MockitoSettings(strictness = Strictness.LENIENT)
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
class ComptabiliteDaoImplTest extends ConsumerHelperTest {

    private static DaoProxyImpl daoProxy;

    private ComptabiliteDaoImpl comptabiliteDao;

    private EcritureComptable ecritureComptableSample;

    @BeforeEach
    public void setUp() {
        daoProxy = (DaoProxyImpl) ConsumerHelperTest.getBean("DaoProxy");
        comptabiliteDao = (ComptabiliteDaoImpl) daoProxy.getComptabiliteDao();

        ecritureComptableSample = new EcritureComptable();
        ecritureComptableSample.setJournal(new JournalComptable("AC", "Achat"));
        ecritureComptableSample.setDate(Date.from(Instant.now()));
        ecritureComptableSample.setLibelle("Libelle");
        ecritureComptableSample.setReference("AC-2019/00001");
    }

    @AfterEach
    public void reInit() {
        ecritureComptableSample = null;
    }


    @Test
    @Order(1)
    void getListCompteComptableTest() {
        List<CompteComptable> compteComptableList;

        compteComptableList = comptabiliteDao.getListCompteComptable();

        assertEquals(7, compteComptableList.size());
    }

    @Test
    @Order(1)
    void getListJournalComptableTest() {
        List<JournalComptable> journalComptableList;

        journalComptableList = comptabiliteDao.getListJournalComptable();

        assertEquals(4, journalComptableList.size());
    }

    @Test
    @Order(2)
    void getListEcritureComptableTest() {
        List<EcritureComptable>  ecritureComptableList = comptabiliteDao.getListEcritureComptable();

        assertEquals(6, ecritureComptableList.size());
    }

    @Test
    @Order(3)
    void getEcritureComptableTest() throws NotFoundException {
        EcritureComptable ecritureComptableSample;

        ecritureComptableSample = comptabiliteDao.getEcritureComptable(-1);

        assertEquals("AC-2016/00001", ecritureComptableSample.getReference());
    }

    @Test
    @Order(4)
    void getEcritureComptableByRefTest() throws NotFoundException {
        EcritureComptable ecritureComptableSample;

        ecritureComptableSample = comptabiliteDao.getEcritureComptableByRef("BQ-2016/00003");

        assertEquals("Paiement Facture F110001", ecritureComptableSample.getLibelle());
    }

    @Test
    @Order(5)
    void loadListLigneEcritureTest() throws NotFoundException {
        EcritureComptable ecritureComptableSample = comptabiliteDao.getEcritureComptable(-1);

        comptabiliteDao.loadListLigneEcriture(ecritureComptableSample);

        assertEquals(3, ecritureComptableSample.getListLigneEcriture().size());
    }

    @Test
    @Order(6)
    void insertEcritureComptableTest() throws NotFoundException {
        comptabiliteDao.insertEcritureComptable(ecritureComptableSample);

        assertEquals(ecritureComptableSample.getReference(), comptabiliteDao.getEcritureComptableByRef(ecritureComptableSample.getReference()).getReference());
    }

    @Test
    @Order(7)
    void insertListLigneEcritureComptableTest() throws NotFoundException {
        ecritureComptableSample = comptabiliteDao.getListEcritureComptable().get(comptabiliteDao.getListEcritureComptable().size() - 1);
        ecritureComptableSample.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(706),
                null, null,
                new BigDecimal(123)));
        ecritureComptableSample.getListLigneEcriture().add(new LigneEcritureComptable(new CompteComptable(706),
                null, null,
                new BigDecimal(123)));

        comptabiliteDao.insertListLigneEcritureComptable(ecritureComptableSample);

        assertEquals(2, ecritureComptableSample.getListLigneEcriture().size());
    }

    @Test
    @Order(8)
    void updateEcritureComptableTest() throws NotFoundException {

        ecritureComptableSample = comptabiliteDao.getEcritureComptableByRef("AC-2016/00001");
        ecritureComptableSample.setLibelle("Nouveau libelle 1234");

        comptabiliteDao.updateEcritureComptable(ecritureComptableSample);

        assertEquals("Nouveau libelle 1234", comptabiliteDao.getEcritureComptableByRef("AC-2016/00001").getLibelle());
    }

    @Test
    @Order(9)
    void deleteEcritureComptableTest() throws NotFoundException {
        EcritureComptable ecritureComptableSample = new EcritureComptable();
        ecritureComptableSample.setJournal(new JournalComptable("AC", "Achat"));
        ecritureComptableSample.setDate(new Date());
        ecritureComptableSample.setLibelle("Libelle");
        ecritureComptableSample.setReference("AC-2019/99999");

        comptabiliteDao.insertEcritureComptable(ecritureComptableSample);

        comptabiliteDao.deleteEcritureComptable(comptabiliteDao.getEcritureComptableByRef("AC-2019/99999").getId());

        assertThrows(NotFoundException.class, () -> comptabiliteDao.getEcritureComptableByRef("AC-2019/99999"));
    }

    @Test
    @Order(10)
    void deleteListLigneEcritureComptableTest() throws NotFoundException {
        comptabiliteDao.insertListLigneEcritureComptable(ecritureComptableSample);

        comptabiliteDao.deleteListLigneEcritureComptable(comptabiliteDao.getEcritureComptableByRef("AC-2019/00001").getId());

        assertEquals(0, comptabiliteDao.getEcritureComptableByRef("AC-2019/00001").getListLigneEcriture().size());
    }
}